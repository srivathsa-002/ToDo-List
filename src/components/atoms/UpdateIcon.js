import React from 'react';
import { Tooltip, IconButton } from "@material-ui/core";
import ArrowUpwardIcon from '@material-ui/icons/ArrowUpward';

export default class UpdateIcon extends React.Component {
    render() {
        return (
            <Tooltip data-testid="update" onClick={this.props.click} title="Update Todo" placement={this.props.placement} arrow>
                <div>
                    <IconButton>
                        <ArrowUpwardIcon />
                    </IconButton>
                </div>
            </Tooltip>
        );
    }
}