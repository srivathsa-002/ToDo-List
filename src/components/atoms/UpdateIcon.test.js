import React from "react";
import {render,fireEvent} from "@testing-library/react";
import UpdateIcon from "./UpdateIcon";
// import { faItalic } from "@fortawesome/free-solid-svg-icons";
import "@testing-library/jest-dom/extend-expect";
import {create} from "react-test-renderer";

describe("UpdateIcon testing", () => {
    test("Matches the snapshot", () => {
      const input = create(<UpdateIcon />);
      expect(input.toJSON()).toMatchSnapshot();
    });
});

it("renders correctly", () => {
    const{getByTestId} = render(<UpdateIcon />)

    expect(getByTestId("update")).toBeTruthy()
    // expect(queryByPlaceholderText("Todo...")).toBeTruthy()
})

// describe("Input value", () => {
//     it("updates on change", () => {
//         const {queryByPlaceholderText} = render(<InputBox />)
//         const inputBox = queryByPlaceholderText("Todo...")
//         fireEvent.change(inputBox, {target: {value: "test"}})
//         expect(inputBox.value).toBe("test")
//     })
// })
