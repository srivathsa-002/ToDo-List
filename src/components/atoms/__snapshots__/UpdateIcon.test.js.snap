// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateIcon testing Matches the snapshot 1`] = `
<div
  aria-describedby={null}
  className=""
  data-testid="update"
  onBlur={[Function]}
  onFocus={[Function]}
  onMouseLeave={[Function]}
  onMouseOver={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  title="Update Todo"
>
  <button
    className="MuiButtonBase-root MuiIconButton-root"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiIconButton-label"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M4 12l1.41 1.41L11 7.83V20h2V7.83l5.58 5.59L20 12l-8-8-8 8z"
        />
      </svg>
    </span>
    <span
      className="MuiTouchRipple-root"
    />
  </button>
</div>
`;
